ouroboros-network/testlib/Test/Ouroboros/Network/Diffusion/Testnet/Cardano.hs

```diff
@@ -2939,9 +2940,12 @@ prop_diffusion_target_active_below ioSimTrace traceNumber =
             . selectDiffusionPeerSelectionEvents
             $ events
 
-          govInProgressPromoteWarmSig :: Signal (Set NtNAddr)
-          govInProgressPromoteWarmSig =
-            selectDiffusionPeerSelectionState Governor.inProgressPromoteWarm  events
+          govInProgressIneligibleSig :: Signal (Set NtNAddr)
+          govInProgressIneligibleSig =
+            selectDiffusionPeerSelectionState
+            (uncurry Set.union . (    Governor.inProgressPromoteWarm
+                                  &&& Governor.inProgressDemoteWarm))
```

Comment:
An alternative is `(,) <$> Governor.inProgressPromoteWarm <*> Governor.inProgressDemoteWarm` for which one needs to be aware that `(->) r` is an `Applicative` - so I am not sure if it's clearer :grin: